docs/components/icons/Example.vue

```diff
@@ -1,31 +1,133 @@
+    : { [current]: icons[current] };
+
+  return acc;
+}, {});
+
+const getIconName = (icon) => icon.replace(/Icon|\d+/g, "");
+
+const setIconData = (fullName, event) => {
+  const reactIconName = Object.keys(reactIcons)
+    .find((icon) => icon === fullName)
+    .replace("Icon", "");
+
+  const outputString = decamelize(reactIconName, { separator: "-" }).replace(
+    /([a-zA-Z])(\d+)/,
+    "$1-$2"
+  );
+
+  modalData = {
+    iconName: fullName,
+    react: `import ${fullName} from '@warp-ds/icons/react/${outputString}';`,
+    reactSyntax: `<${fullName} />`,
+    vue: `import ${fullName} from '@warp-ds/icons/vue/${outputString}';`,
+    vueSyntax: `<icon-${outputString} />`,
+    elements: `import from '@warp-ds/icons/elements/${outputString}';`,
+    elementsSyntax: `<w-icon-${outputString}></w-icon-${outputString}>`,
+  };
+};
+
+const reset = () => {
+  modalData = {
+    iconName: "",
+    react: "",
+    reactSyntax: "",
+    vue: "",
+    vueSyntax: "",
+    elements: "",
+    elementsSyntax: "",
+  };
+};
 </script>
 
 <template>
-  <section v-for="size in Object.keys(mappedIconsBySize)" :key="size" style="background-color: var(--vp-c-bg-alt);" class="rounded-8 p-24 mb-24">
+  <section
+    v-for="size in Object.keys(mappedIconsBySize)"
+    :key="size"
+    style="background-color: var(--vp-c-bg-alt)"
+    class="rounded-8 p-24 mb-24"
+  >
     <h2 style="color: var(--vp-c-text-1)">{{ size }}</h2>
+    <w-modal
+      v-if="modalData"
+      title="Icon usage"
+      :right="{ 'aria-label': 'Close' }"
+      @dismiss="
+        showModal = false;
+        reset();
+      "
+      v-model="showModal"
+      @right="
+        showModal = false;
+        reset();
+      "
+    >
+      <div>
+        <h1 class="h4 mb-16">
+          You can use the following import for icon: {{ modalData.iconName }}
+        </h1>
+        <p>
+          Usage for React:
+          <div class="border rounded-8 my-8 p-8">{{ modalData.react }}</div>
+          <div class="border rounded-8 my-8 p-8">{{ modalData.reactSyntax}}</div>
+        </p>
+        <p>
+          Usage for Vue:
+          <div class="border rounded-8 my-8 p-8">{{ modalData.vue }}</div>
+          <div class="border rounded-8 my-8 p-8">{{ modalData.vueSyntax}}</div>
+        </p>
+        <p>
+          Usage for Elements:
+          <div class="border rounded-8 my-8 p-8">{{ modalData.elements }}</div>
+          <div class="border rounded-8 my-8 p-8">{{ modalData.elementsSyntax}}</div>
+        </p>
+      </div>
+    </w-modal>
     <main class="max-w-screen-xl mx-auto px-32">
       <div class="grid gap-24 grid-cols-minmax-100px">
-        <div v-for="(icon, fullName) in mappedIconsBySize[size]" :key="fullName">
+        <div
+          class="cursor-pointer"
+          @click="
+            showModal = true;
+            setIconData(fullName, event);
+          "
+          v-for="(icon, fullName) in mappedIconsBySize[size]"
+          :key="fullName"
+        >
```

Comment:
Thinking of accessibility here. Would it make sense to use a button element here? Alternatively, we could let assistive technologies know that it behaves like a button by setting a `role="button"` here. Then again we'll need to allow users to tab to these interactive elements, so you'd need to add `tabindex="0"`, too. You'd get all that for free if you used a button 😄

Reply:
From what they said at this morning's accessibility workshop, they advised to use a button element instead of a div with a @click :)

Reply:
Done ✅